src/backend/main.mo

```diff
@@ -466,4 +504,99 @@ actor {
     return "Hello, " # name # "! " # "Your PrincipalId is: " # Principal.toText(caller);
   };
 
+  // Images
+
+  public shared (msg) func uploadImg(imgId : ImgId, image : Blob) {
+    storeBlobImg(imgId, image);
+  };
+
+  public query ({ caller }) func getPic(id : ImgId) : async Blob {
+    var pic = loadBlobImg(id);
+    switch (pic) {
+      case (null) {
+        return Blob.fromArray([]);
+      };
+      case (?existingPic) {
+        return existingPic;
+      };
+    };
+  };
+
+  private func storeBlobImg(imgId : ImgId, value : Blob) {
+    var size : Nat = Nat32.toNat(Nat32.fromIntWrap(value.size()));
+    // Each page is 64KiB (65536 bytes)
+    var growBy : Nat = size / 65536 + 1;
+    let a = Memory.grow(Nat64.fromNat(growBy));
+    Memory.storeBlob(currentMemoryOffset, value);
+    imgOffset.put(imgId, currentMemoryOffset);
+    imgSize.put(imgId, size);
+    size := size + 4;
+    currentMemoryOffset += Nat64.fromNat(size);
+  };
+
+  private func loadBlobImg(imgId : ImgId) : ?Blob {
```

Comment:
tieto funkcie by asi mohli byť skryté vo vlastnom module, nie? keď sú to len utility funkcie pre prácu s obrázkami a nie sú priamo volané